auth/src/main/java/com/farsitel/bazaar/storage/connection/ServiceStorageConnection.kt

```diff
@@ -0,0 +1,181 @@
+package com.farsitel.bazaar.storage.connection
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.Bundle
+import android.os.IBinder
+import androidx.lifecycle.LifecycleOwner
+import com.farsitel.bazaar.BAZAAR_PACKAGE_NAME
+import com.farsitel.bazaar.BazaarResponse
+import com.farsitel.bazaar.storage.InAppBazaarStorage
+import com.farsitel.bazaar.storage.callback.BazaarStorageCallback
+import com.farsitel.bazaar.thread.BackgroundThread
+import com.farsitel.bazaar.thread.MainThread
+import com.farsitel.bazaar.util.AbortableCountDownLatch
+import com.farsitel.bazaar.util.ext.toBase64
+
+internal class ServiceStorageConnection(
+    private val context: Context,
+    private val backgroundThread: BackgroundThread,
+    private val mainThread: MainThread
+) : StorageConnection(context), ServiceConnection {
+
+    private var storageService: InAppBazaarStorage? = null
+    private var connectionThreadSecure: AbortableCountDownLatch? = null
+
+    fun connect(): Boolean {
```

Comment:
This logic here is similar to `ServiceAuthConnection`, You can handle that in a better way in order to have DRY principles(Don't Repeat Yourself)